docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,810 @@
+
+    <!-- Package the Newtonsoft.Json dependency alongside the generator assembly -->
+    <None Include="$(PkgNewtonsoft_Json)\lib\netstandard2.0\*.dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+  </ItemGroup>
+</Project>
+```
+
+```C#
+[Generator]
+public class JsonUsingGenerator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var pipeline = context.AdditionalTextsProvider.Select(static (text, cancellationToken) =>
+        {
+            if (!text.Path.EndsWith("*.json"))
+            {
+                return null;
+            }
+
+            return (Name: text.Name, Value: Newtonsoft.Json.JsonConvert.DeserializeObject<MyObject>(text.GetText(cancellationToken)));
+        })
+        .Where((pair, ct) => pair is not null);
+
+        context.RegisterSourceOutput(pipeline, static (context, pair) =>
+        {
+            var sourceText = SourceText.From($$"""
+                namespace GeneratedNamespace
+                {
+                    public class GeneratedClass
+                    {
+                        public static const (int A, int B) SerializedContent = ({{pair.A}}, {{pair.B}});
+                    }
+                }
+                """, Encoding.UTF8);
+
+            context.AddSource($"{pair.Name}generated.cs", sourceText)
+        });
+    }
+
+    record MyObject(int A, int B);
+}
+```
+
+### Access Analyzer Config properties
+
+**User Scenarios:**
+
+- As a generator author I want to access the analyzer config properties for a syntax tree or additional file.
+- As a generator author I want to access key-value pairs that customize the generator output.
+- As a user of a generator I want to be able to customize the generated code and override defaults.
+
+**Solution**: Generators can access analyzer config values via the `AnalyzerConfigOptionsProvider`. Analyzer config values can either be accessed in the context of a `SyntaxTree`, `AdditionalFile` or globally via `GlobalOptions`. Global options are 'ambient' in that they don't apply to any specific context, but will be included when requesting option within a specific context.
+
+Note that this is one of the few cases that it is necessary to put a `SyntaxNode` into a pipeline, as you need the tree in order to get the generator option.
+Try to get the `SyntaxNode` out of the pipeline as fast as possible to avoid making the model not correctly equatable.
+
+A generator is free to use a global option to customize its output. For example, consider a generator that can optionally emit logging. The author may choose to check the value of a global analyzer config value in order to control whether or not to emit the logging code. A user can then choose to enable the setting per project via an `.editorconfig` file:
+
+```.editorconfig
+mygenerator_emit_logging = true
+```
```

Comment:
I think this is something confusing. AFAIK, `editorconfig`s are per-tree and `globalconfig`s are per-project. Did this mean to say `.globalconfig` file?

Reply:
Probably. I copied it from the old docs: @chsienki, does that doc need to be updated as well?